[1.18] Mappings for `net.minecraft.client.gui`

geosskk opened the pull request.

geosskk committed:
[1.18] Mappings for `net.minecraft.client.gui`.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/ParchmentMC/Parchment?pullRequest=116) <br/>All committers have signed the CLA.

sciwhiz12 commented on data/net/minecraft/client/gui/Font.mapping:
```diff
@@ -182,6 +183,17 @@ CLASS net/minecraft/client/gui/Font
+		ARG 7 buffer
+		ARG 8 isTransparent
+		ARG 9 colorBackground
+		ARG 10 packedLight
```
These should follow the parameters names in the constructor (and method in the case for `colorBackground`/`backgroundColor`) for `StringRenderOutput`.
```suggestion
	METHOD renderText (Lnet/minecraft/util/FormattedCharSequence;FFIZLcom/mojang/math/Matrix4f;Lnet/minecraft/client/renderer/MultiBufferSource;ZII)F
		ARG 1 processor
		ARG 2 x
		ARG 3 y
		ARG 4 color
		ARG 5 dropShadow
		ARG 6 pose
		ARG 7 bufferSource
		ARG 8 seeThrough
		ARG 9 backgroundColor
		ARG 10 packedLightCoords
```

sciwhiz12 commented on data/net/minecraft/client/gui/screens/multiplayer/WarningScreen.mapping:
```diff
@@ -0,0 +1,9 @@
+CLASS net/minecraft/client/gui/screens/multiplayer/WarningScreen
+	METHOD <init> (Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/client/gui/screens/Screen;)V
+		ARG 1 title
+		ARG 2 content
+		ARG 3 checkbox
+		ARG 4 narration
+		ARG 5 lastScreen
```
Parameter names which are assigned directly to fields should be named the same as the field.
```suggestion
	METHOD <init> (Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/Component;Lnet/minecraft/client/gui/screens/Screen;)V
		ARG 1 titleComponent
		ARG 2 content
		ARG 3 check
		ARG 4 narration
		ARG 5 previous
```

sciwhiz12 commented on data/net/minecraft/client/gui/screens/recipebook/GhostRecipe.mapping:
```diff
@@ -8,6 +8,8 @@ CLASS net/minecraft/client/gui/screens/recipebook/GhostRecipe
 	METHOD render (Lcom/mojang/blaze3d/vertex/PoseStack;Lnet/minecraft/client/Minecraft;IIZF)V
 		ARG 1 poseStack
 		ARG 2 minecraft
+		ARG 3 leftPos
+		ARG 4 topPos
```
What would be better here: these names, or `x` and `y`?

geosskk replied:
I'm not too sure, on the one hand it seems like `x` and `y` would be simpler, however every time this method is called the name of the fields that are used are `leftPos` and `topPos`.

For example, in `net/minecraft/client/gui/screens/inventory/CraftingScreen.java` the `render` method uses `this.leftPos` and `this.topPos`. _(This naming also occurs in `AbstractFurnaceScreen.java` and `InventoryScreen.java`)_
![image](https://user-images.githubusercontent.com/32211101/162414963-1472e258-7daa-4601-a668-d1b3334931e3.png)

_The method `renderGhostRecipe` just calls the method `render` from `net/minecraft/client/gui/screens/recipebook/GhostRecipe.java`._
![image](https://user-images.githubusercontent.com/32211101/162415320-4a39a660-368f-4a94-af49-a54857d6d924.png)

sciwhiz12 commented on data/net/minecraft/client/gui/screens/worldselection/CreateWorldScreen.mapping:
```diff
@@ -12,11 +12,21 @@ CLASS net/minecraft/client/gui/screens/worldselection/CreateWorldScreen
 		ARG 0 lastScreen
 	METHOD createFromExisting (Lnet/minecraft/client/gui/screens/Screen;Lnet/minecraft/server/WorldStem;Ljava/nio/file/Path;)Lnet/minecraft/client/gui/screens/worldselection/CreateWorldScreen;
 		ARG 0 lastScreen
+		ARG 1 worldstem
```
Camel-case to match the class name:
```suggestion
		ARG 1 worldStem
```

sciwhiz12 commented on data/net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent.mapping:
```diff
@@ -31,5 +33,7 @@ CLASS net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent
 	METHOD toString (Ljava/util/OptionalLong;)Ljava/lang/String;
 		ARG 0 seed
+	METHOD updateDataPacks (Lnet/minecraft/server/WorldStem;)V
+		ARG 1 worldstem
```
Camel-case to match the class name:
```suggestion
		ARG 1 worldStem
```

sciwhiz12 commented on data/net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent.mapping:
```diff
@@ -20,6 +20,8 @@ CLASS net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent
 	METHOD lambda$init$9 (Lnet/minecraft/client/Minecraft;Lnet/minecraft/client/gui/screens/worldselection/CreateWorldScreen;Lnet/minecraft/core/RegistryAccess$Writable;Lcom/mojang/serialization/Lifecycle;Lnet/minecraft/world/level/levelgen/WorldGenSettings;)V
 		ARG 5 settings
+	METHOD makeSettings (Z)Lnet/minecraft/world/level/levelgen/WorldGenSettings;
+		ARG 1 isHardcore
```
Could simplify the name to `hardcode`:
```suggestion
		ARG 1 hardcode
```

sciwhiz12 commented on data/net/minecraft/client/gui/screens/worldselection/WorldSelectionList.mapping:
```diff
@@ -10,9 +10,15 @@ CLASS net/minecraft/client/gui/screens/worldselection/WorldSelectionList
 	METHOD moveSelection (Lnet/minecraft/client/gui/components/AbstractSelectionList$SelectionDirection;)V
 		ARG 1 ordering
+	METHOD refreshList (Ljava/util/function/Supplier;Z)V
+		ARG 1 supplier
```
A more descriptive name would be something like `searchStringSupplier` or `filterStringSupplier`, as the supplied string is used to filter out any levels which do not contain the string as a substring of their level name or level ID.

sciwhiz12 commented on data/net/minecraft/client/gui/screens/worldselection/WorldSelectionList.mapping:
```diff
@@ -10,9 +10,15 @@ CLASS net/minecraft/client/gui/screens/worldselection/WorldSelectionList
 	METHOD setSelected (Lnet/minecraft/client/gui/screens/worldselection/WorldSelectionList$WorldListEntry;)V
 		ARG 1 entry
 	CLASS WorldListEntry
+		METHOD <init> (Lnet/minecraft/client/gui/screens/worldselection/WorldSelectionList;Lnet/minecraft/client/gui/screens/worldselection/WorldSelectionList;Lnet/minecraft/world/level/storage/LevelSummary;)V
+			ARG 2 worldSelectionList
+			ARG 3 levelSummary
```
Parameter names which are assigned directly to fields should be named the same as the field.
```suggestion
		METHOD <init> (Lnet/minecraft/client/gui/screens/worldselection/WorldSelectionList;Lnet/minecraft/client/gui/screens/worldselection/WorldSelectionList;Lnet/minecraft/world/level/storage/LevelSummary;)V
			ARG 2 worldSelectionList
			ARG 3 summary
```

geosskk committed:
Add most of the requested changes

geosskk committed:
Update WorldSelectionList.mapping

Changed the parameter supplier from the method refreshList to filterLevelSupplier as the supplied string is used to filter out levels which their names or ids do not contain the supplied string.

Also added the parameter for ARG 8 in the constructor for WorldSelectionList to match the parameter from the method refreshList.
